lib/mapper.js

```diff
@@ -72,13 +72,21 @@ exports.track = function(track) {
       }
     });
   } else {
+    if (typeof track.proxy('context.traits.email') !== 'undefined'){
```

Comment:
Rather than reading and checking `context.traits.email` directly, this should use `track.email()`.